### PR TITLE
add types in auth-reducer.tsx

### PR DESCRIPTION
**Pull request by LuydmilaKolesnikova**

### CODE REVIEW

File: src/redux/auth-reducer.tsx [hunk 1]
```diff
@@ -79,7 +116,7 @@ export const registering = (login, email, password) => {
 };
 
 export const logout = () => {
-  return async (dispatch) => {
+  return async (dispatch: any) => {
```
Comment (quanterion) [inline src/redux/auth-reducer.tsx]: dispatch never used. Either remove it or use `dispatch: unknown`

File: src/redux/auth-reducer.tsx [hunk 2]
```diff
@@ -3,29 +3,66 @@ import { authAPI } from "../api/authAPI";
 const SET_USER_DATA = "SET_USER_DATA";
 const REGISTRATION_SUCCESS = "REGISTRATION_SUCCESS";
 
-export const setAuthUserData = (id, email, login, isAuth) => {
+type payloadType = {
+  id: null | number;
```
Comment (quanterion) [inline src/redux/auth-reducer.tsx]: usually it's better to use undefined for values that can be omitted and write `id: number | undefined;` that can be shorted to just `id: number?;`

File: src/redux/auth-reducer.tsx (hunk 1 above)
Comment (LuydmilaKolesnikova) [reply src/redux/auth-reducer.tsx]: dispatch never used? What do you mean? 

`export const logout = () => {
  return async (**dispatch**: any) => {
    try {
      await authAPI.logout();
      **dispatch**(setAuthUserData(null, null, null, false));
    } catch (e) {
      alert("No response from server");
    }
  };
};`

File: src/redux/auth-reducer.tsx [hunk 3]
```diff
@@ -3,29 +3,66 @@ import { authAPI } from "../api/authAPI";
 const SET_USER_DATA = "SET_USER_DATA";
 const REGISTRATION_SUCCESS = "REGISTRATION_SUCCESS";
 
-export const setAuthUserData = (id, email, login, isAuth) => {
+type payloadType = {
```
Comment (quanterion) [inline src/redux/auth-reducer.tsx]: `type payloadType = {` -> `interface PayloadData { ` https://www.typescriptlang.org/docs/handbook/interfaces.html

File: src/redux/auth-reducer.tsx [hunk 4]
```diff
@@ -3,29 +3,66 @@ import { authAPI } from "../api/authAPI";
...
+  success: boolean
+): registrationSuccessActionType => {
   return {
     type: REGISTRATION_SUCCESS,
     success,
   };
 };
 
-let initialState = {
+let state = {
```
Comment (quanterion) [inline src/redux/auth-reducer.tsx]: do not declare types inline, use interfaces and reuse them

File: src/redux/auth-reducer.tsx [hunk 5]
```diff
@@ -3,29 +3,66 @@ import { authAPI } from "../api/authAPI";
...
+  isAuth: boolean
+): SetUserDataActionType => {
   return {
     type: SET_USER_DATA,
     payload: { id, email, login, isAuth },
   };
 };
 
-export const registrationSuccess = (success) => {
+type registrationSuccessActionType = {
```
Comment (quanterion) [inline src/redux/auth-reducer.tsx]: `registrationSuccessActionType` -> `RegistrationSuccessAction` all types UpperCased, variables - lowerCased.
no need to add word 'type' to the name of each type, it is useless

File: src/redux/auth-reducer.tsx [hunk 6]
```diff
@@ -3,29 +3,66 @@ import { authAPI } from "../api/authAPI";
...
+  type: typeof REGISTRATION_SUCCESS;
+  success: boolean;
+};
+
+export const registrationSuccess = (
```
Comment (quanterion) [inline src/redux/auth-reducer.tsx]: export function registrationSuccess(...){ ...}

File: src/redux/auth-reducer.tsx [hunk 7]
```diff
@@ -3,29 +3,60 @@ import { authAPI } from "../api/authAPI";
...
-};
+}
+
+interface InitialState extends PayloadData {
+  isRegistrationSuccess?: boolean;
+}
 
-let initialState = {
+let initialState: InitialState = {
   id: null,
```
Comment (quanterion) [inline src/redux/auth-reducer.tsx]: no need to declare id: null, just omit it at all

File: src/redux/auth-reducer.tsx [hunk 8]
```diff
@@ -44,19 +75,19 @@ const authReducer = (state = initialState, action) => {
...
 };
 
-export const getAuthUserData = () => {
+export function getAuthUserData() {
   return async (dispatch) => {
     try {
       const response = await authAPI.me();
       let { id, email, login } = response.data;
-      dispatch(setAuthUserData(id, email, login, id ? true : false));
+      dispatch(setAuthUserData(id ? true : false, id, email, login));
```
Comment (quanterion) [inline src/redux/auth-reducer.tsx]: if you can compute isAuth from id, why not do in function itself and remove isAuth parameter

File: src/redux/header-reducer.tsx [hunk 9]
```diff
@@ -2,45 +2,71 @@ import { headerAPI } from "../api/headerAPI";
...
-    messages: null,
-    notifications: null,
-  },
+    followfeed?: number;
+    messages?: number;
+    notifications?: number;
+  };
+}
+
+interface InitialState extends ProfileData, Statistics, NewActions {}
```
Comment (quanterion) [inline src/redux/header-reducer.tsx]: `interface  InitialState {
  newActions: NewActions;
 ...
}`
and remove newAction from NewAction, statistics from Statistics, etc

File: src/redux/header-reducer.tsx [hunk 10]
```diff
@@ -2,45 +2,71 @@ import { headerAPI } from "../api/headerAPI";
...
       };
     }
     default:
       return state;
   }
 };
 
-export const getAuthUserData = (id) => {
+export const getAuthUserData = (id: number) => {
   return async (dispatch) => {
```
Comment (quanterion) [inline src/redux/header-reducer.tsx]: add dispatch type

File: src/api/authAPI.tsx [hunk 11]
```diff
@@ -1,32 +1,32 @@
 import { instance } from "./instanceAPI";
 
 export const authAPI = {
-  me() {
+  me(): any {
```
Comment (quanterion) [inline src/api/authAPI.tsx]: 1. don not use any, specify concrete type or unknown

File: src/api/authAPI.tsx [hunk 12]
```diff
@@ -1,32 +1,32 @@
...
     return instance.get(`auth`).then((response) => response);
```
Comment (quanterion) [inline src/api/authAPI.tsx]: looks like `.then((response) => response)` doing nothing

File: src/api/authAPI.tsx [hunk 13]
```diff
@@ -1,32 +1,32 @@
...
-        password: password,
+        login,
+        email,
+        password,
       })
       .then((response) => response);
   },
 
-  logout() {
+  logout(): any {
```
Comment (quanterion) [inline src/api/authAPI.tsx]: same here
avoid any

File: src/redux/auth-reducer.tsx [hunk 14]
```diff
@@ -3,29 +3,57 @@ import { authAPI } from "../api/authAPI";
...
+  type: typeof SET_USER_DATA;
+  payload: PayloadData;
+}
+
+export function setAuthUserData(
+  id?: number,
+  email?: string,
+  login?: string
+): SetUserDataAction {
+  const isAuth = id ? true : false;
```
Comment (quanterion) [inline src/redux/auth-reducer.tsx]: do you need isAuth in payload if it can be simply computed

File: src/api/authAPI.tsx [hunk 15]
```diff
@@ -1,38 +1,40 @@
 import { instance } from "./instanceAPI";
 
+type MeResponse = { data?: { id: number; email: string; login: string } };
+
 export const authAPI = {
-  me() {
-    return instance.get(`auth`).then((response) => response);
+  me(): Promise<MeResponse> {
+    return instance.get(`auth`).then();
```
Comment (quanterion) [inline src/api/authAPI.tsx:7]: .then not need at all

**Pull request merged**